File: Algorithm/count_and_say.py
class Solution:
    def countAndSay(self,n):
        word='1'
        for x in range(n-1):
            pre=''
            stemp=''
            num=1
            for element in word:
                if element==pre:
                    num+=1
                elif pre!='':
                    stemp+=str(num)+pre
                    num=1
                pre=element
            stemp+=str(num)+pre
            word=stemp                 
        return word
        

    
if __name__=='__main__':
    sol=Solution()
    sol.countAndSay(2)
    for x in range(1,10):
        print sol.countAndSay(x)

File: Algorithm/CPP/count_and_say.cpp
# include <stdio.h>
# include <iostream>
# include <vector>
# include <string>
# include <cctype>
# include <algorithm>
using namespace std;
class Solution {

public:
	string getNumber(string s)
	{

		string res;

		char pre = s[0];
		char ch = pre;
		int count = 0;
		for (auto chr:s)
		{
			ch = chr;
			if (ch != pre)
			{
				res += to_string(count) + pre;
				count = 1;
				pre = ch;
			}
			else
			{
				count += 1;
			}
		}
		res += to_string(count) + ch;
		return res;

	}
	string countAndSay(int n)
	{
		string  number = "1";
		for (auto i = 0; i < n-1; ++i)
		{
			number =getNumber(number);
		}
		return number;
	}
};
int  main()
{
	Solution sol;
	int n = 40;
	//cout << char(2 + '0') << endl;
	cout << sol.countAndSay(n) << endl;
	cin.get();
	return 0;
}
